src/source-verifier/tact-source-verifier.ts

```diff
@@ -75,11 +75,15 @@ export class TactSourceVerifier implements SourceVerifier {
           (filenameA.endsWith(".tact") ? 1 : 0) - (filenameB.endsWith(".tact") ? 1 : 0),
       );
 
+      const compiledHash = Cell.fromBoc(Buffer.from(v.package.code, "base64"))[0]
+        .hash()
+        .toString("base64");
+
       return {
         compilerSettings,
         error: null,
-        hash: Cell.fromBoc(Buffer.from(v.package.code, "base64"))[0].hash().toString("base64"),
-        result: "similar",
+        hash: compiledHash,
+        result: compiledHash === payload.knownContractHash ? "similar" : "not_similar",
```

Comment:
This is an important, unrelated fix -> prior to this tact only verified the integrity of the pkg file but did not compare to the onchain contract being validated